src/main/java/ben/Main.java

```diff
@@ -0,0 +1,67 @@
+package ben;
+
+import ben.commands.Command;
+import ben.exceptions.BenException;
+import ben.parser.Parser;
+import ben.storage.Storage;
+import ben.tasks.TaskList;
+import ben.ui.Ui;
+
+import java.io.FileNotFoundException;
+
+/**
+ * The main class for the Ben task management application.
+ */
+public class Main {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs the Main object with the specified file path for storage.
+     *
+     * @param filePath The file path for storage.
+     */
+    public Main(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (BenException | FileNotFoundException e) {
+            ui.showError(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the Ben task management application.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
```

Comment:
Perhaps using a more meaningful name then c even if it is only used in the next line